analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -607,7 +607,54 @@ End Sub
         }
 
         [TestMethod]
-        public void FieldReference_Read_SetsNotNull()
+        public void Invocation_ValidatedNotNullAttribute_SetsNotNullOnArgumentsMarkedWithAttribute()
+        {
+            var code = @"
+using System;
+
+public class Sample
+{
+    public void Main(object o1, object o2, object o3, object o4, object o5)
+    {
+        Guard.NotNullExt(o1);
+        o2.NotNullExt();
+        NotNullInst(o3);
+        NotNullInst(o4, o5);
+        Tag(""AfterGuard_o1"", o1);
+        Tag(""AfterGuard_o2"", o2);
+        Tag(""AfterGuard_o3"", o3);
+        Tag(""AfterGuard_o4"", o4);
+        Tag(""AfterGuard_o5"", o5);
+    }
+
+    private void NotNullInst([ValidatedNotNullAttribute] object value)
+    {
+        // Skip implementation to make sure, the attribute is driving the constraint
+        // _ = value ?? throw new ArgumentNullException();
+    }
+
+    private void NotNullInst([ValidatedNotNullAttribute] object value1, [ValidatedNotNullAttribute] object value2) { }
+
+    private void Tag(string name, object arg) { }
```

Comment:
_Rebase conflict:_ This will need to be static after #6060, right? So we'll need to rebase not to break master